source/extensions/common/wasm/wasm.h

```diff
@@ -213,6 +241,18 @@ class Context : public Http::StreamFilter,
   virtual void httpRespond(const Pairs& response_headers, absl::string_view body,
                            const Pairs& response_trailers);
 
+  // Stats/Metrics
+  enum class MetricType : uint32_t {
```

Comment:
I am not sure why enum is required. Even so we should be able to reuse the value above
```
static const uint32_t kMetricTypeCounter = 0x0;

```

Reply:
The enum is a duplicate of one in the API and it is supposed to be opaque. The implementation uses the same numbers to encode the type of the metric into the metric_id.   Yes, they could be reused, but then it would require a bunch of casting .  How about I add a comment and some static_assert(s) to ensure that they remain the same.